dlib/media/ffmpeg_muxer.h

```diff
@@ -886,7 +886,51 @@ namespace dlib
         inline AVSampleFormat   muxer::sample_fmt()             const noexcept { return st.encoder_audio.sample_fmt(); }
         inline AVCodecID        muxer::get_audio_codec_id()     const noexcept { return st.encoder_audio.get_codec_id(); }
         inline std::string      muxer::get_audio_codec_name()   const noexcept { return st.encoder_audio.get_codec_name(); }
+
+// ---------------------------------------------------------------------------------------------------
+
+        template <typename image_type>
+        std::enable_if_t<is_image_type<image_type>::value, void>
+        save_frame(
+            const image_type& image,
+            const std::string& file_name,
+            const std::string& codec_name,
+            const std::unordered_map<std::string, std::string>& codec_options = {}
+        )
+        {
+            const auto codecs = list_codecs();
+            const auto p = std::find_if(codecs.begin(), codecs.end(), [&](const auto& codec){ return codec_name == codec.codec_name; });
+
+            if (p == codecs.end())
+                throw error("ffmpeg::save_frame: codec " + codec_name + " not found");
+
+            if (!p->supports_encoding)
+                throw error("ffmpeg::save_frame: codec " + codec_name + " does not support encoding");
+
+            muxer writer([&] {
+                muxer::args args;
+                args.filepath = file_name;
+                args.enable_image = true;
+                args.enable_audio = false;
+                args.args_image.codec_name = codec_name;
+                args.args_image.h = num_rows(image);
+                args.args_image.w = num_columns(image);
+                args.args_image.framerate = 1;
```

Comment:
You could add here if you want:
```
args.args_image.fmt = pix_traits<pixel_type_t<image_type>>::fmt;
```
This will attempt to use the same pixel format in the image file as the input image. If not supported, doesn't matter, it will automatically convert to something that the image container supports.